File: Leetcode/leetcode#14.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        # Compare prefix at the same address
        # O(len_of_string * num_of_string)
        # We only have to do with the max. and min element in a lit, that's why choosing min, max.
    """
    # Method 1
        if not strs: return ""
        
        s1 = min(strs)
        s2 = max(strs)

        for index, element in enumerate(s1):
            if element != s2[index]:
                return s1[:index]
        return s1 
    """
        if not strs: return ""
        # zip: convert str to list
        ss = list(map(set, zip(*strs)))
        res = ""
        for index, element in enumerate(ss):
            x = list(x)
            if len(x) > 1:
                break
            res = res + x[0]
        return res

File: Leetcode/leetcode#14.c
char * longestCommonPrefix(char ** strs, int strsSize){
    if (!strsSize)
        return "";
    
    char* ans = strs[0];
    // Compare each word with ans
    for(int i=1; i < strsSize; i++)
    {
        char* p = ans;
        char* temp = strs[i];
        while(*p != '\0' && *temp != '\0')
        {
            if(*p != *temp)
            { // If not same, change ans
                *p = '\0';
                break;
            }
            p++;
            temp++;
        }
        // If string is running out, point current string to ans.
        if(*temp == '\0')
        ans = strs[i];
    }
    return ans;
}